File: leetcode/0136-single-number/solution.cpp
class Solution {
public:
    /**
     * RUNTIME: 71 ms (16.39%)
     * MEMORY: 20.1 MB (20.29%)
    */

    int singleNumber(vector<int>& nums) {
        unordered_map<int, int> hashset;
        for (int i : nums) {
            ++hashset[i];
        }
        for (auto each : hashset) {
            if (each.second == 1) {
                return each.first;
            }
        }
        return -1;
    }
};

File: leetcode/0136-single-number/solution.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        """
        RUNTIME: 394 ms (17.4%)
        MEMORY: 16.5 MB (93.88%)
        """
        if len(nums) == 1: return nums[0]
        hashset = {}
        for num in nums:
            if num not in hashset:
                hashset[num] = 1
            else:
                hashset[num] += 1
        for i in hashset:
            if hashset[i] == 1:
                return i
